src/PackageWatcher.ts

```diff
@@ -53,15 +57,7 @@ export class PackageWatcher {
         );
         watcher.onDidCreate(async () => await this.handlePackageChange());
         watcher.onDidChange(async () => await this.handlePackageChange());
-        watcher.onDidDelete(async () => {
-            if (await pathExists(this.workspaceRoot, 'Package.swift')) {
-                // Recreate Package.resolved.
-                this.handlePackageChange();
-            } else {
-                contextKeys.hasPackage = false;
-                contextKeys.packageHasDependencies = false;
-            }
-        });
+        watcher.onDidDelete(async () => await this.handlePackageChange());
```

Comment:
With these changes, both methods are nearly identical, safe for the file name, so we can merge them and add the file name as a parameter.

Reply:
I am looking to have them call different functions, as updating `Package.swift` requires a reload of `swift package describe` while updating `Package.resolved` doesn't. Also the work LLDB work I've got in the pipeline requires functions to be called only on `Package.swift` update.

Reply:
OK, so I'll just leave them as-is?